astropy/units/astrophys.py

```diff
@@ -70,15 +70,16 @@
          doc="Electron mass",
          format={'latex': r'M_{e}', 'unicode': 'Mₑ'})
 # Unified atomic mass unit
-def_unit(['u', 'Da', 'Dalton'], 1.6605387e-27 * si.kg, namespace=_ns,
+def_unit(['u', 'Da', 'Dalton'], _si.u.value * si.kg, namespace=_ns,
          prefixes=True, exclude_prefixes=['a', 'da'],
          doc="Unified atomic mass unit")
 
 ##########################################################################
 # ENERGY
 
-def_unit(['Ry', 'rydberg'], 13.605692 * si.eV, namespace=_ns,
-         prefixes=True,
+def_unit(['Ry', 'rydberg'],
+         _si.Ryd.value * _si.c.value * _si.h.value / _si.e.value * si.eV,
+         namespace=_ns, prefixes=True,
```

Comment:
Oops, sorry, messages clearly crossed (wishing browser would update faster!). Here, the question is really which units are derived and which are measured. For the rydberg, the [search result] (http://physics.nist.gov/cgi-bin/cuu/Results?search_for=rydberg) strongly suggests the value in `eV` is in fact calculated. In fact, its fraction precision is about a factor 1000 better in inverse meters... I wish `u.Ryd` was either not introduced at all, or, if it was, at least defined in inverse meters...

Anyway, following on your latest post, yes, we may have to have constants defined in different units -- I like the idea of overriding `to` to allow that -- let me note that in #3843...